VideoWeb/VideoWeb/ClientApp/src/app/shared/header/header.component.spec.ts

```diff
@@ -36,3 +42,32 @@ describe('HeaderComponent', () => {
         expect(router.navigate).toHaveBeenCalledWith([component.topMenuItems[0].url]);
     });
 });
+
+describe('Header component template file', () => {
+    let fixture: ComponentFixture<HeaderComponent>;
+    let component: HeaderComponent;
+    let debugElement: DebugElement;
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            imports: [RouterTestingModule],
+            declarations: [HeaderComponent, TranslatePipeMock]
+        }).compileComponents();
+    });
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(HeaderComponent);
+        component = fixture.componentInstance;
+        debugElement = fixture.debugElement;
+    });
+
+    it('logout button should route to logout page', () => {
+        component.loggedIn = true;
+
+        fixture.detectChanges();
+
+        const logoutButton = debugElement.query(By.css('.govuk-header__link'));
```

Comment:
Give it an id and use that rather than class.

Reply:
Also a test to check that it's not there when logged out.

Reply:
I will add the extra test, however in TDD you dont normally have to cover existing implementation as you always write a failing test first. This behavior should also be well covered by the acceptance tests.